cmd/bucket-replication.go

```diff
@@ -284,12 +284,24 @@ func filterReplicationStatusMetadata(metadata map[string]string) map[string]stri
 
 type replicationState struct {
 	// add future metrics here
-	replicaCh chan ObjectInfo
+	replicaCh chan replicationMsg
+}
+
+// replicationMsg holds relevant info on object to be replicated.
+type replicationMsg struct {
+	Bucket    string
+	Object    string
+	VersionID string
 }
```

Comment:
I would avoid this struct since this struct lacks information, information i.e carried by ObjectInfo is willy fully lost. This information is helpful in the future so we should keep it. as a struct like this

```
type replicationObjInfo struct {
       ObjectInfo
}
```

Memory optimization might not be necessary IMO.

Reply:
Only these three pieces of info is really required for replicateObject, because it anyway fetches the ObjectInfo from disk with GetObjectNInfo. If the replica channel has a small buffer, all messages get dropped when the buffer is full leaving replication to future crawler runs. With each run picking only 1/16th of the namespace this can lead to significant slowness unless we buffer enough entries.

Reply:
Yeah but I don't think this PR warrants a lot of changes, it's only two things you need to fix https://github.com/minio/minio/pull/10632#issuecomment-703858441

Reply:
ok,done